### ISSUE TITLE
Failure when snapshot contains a variable referring to a list

### ISSUE THREAD

**Issue by alexmojaki** [15r10nk/inline-snapshot#148]
```python
from inline_snapshot import snapshot


def test_foo():
    x = [1]
    assert x == snapshot(x)
```

```
============================================================================= test session starts =============================================================================
platform darwin -- Python 3.12.6, pytest-8.3.4, pluggy-1.5.0
rootdir: /Users/alex
plugins: recording-0.13.2, logfire-2.6.2, anyio-4.3.0, django-4.9.0, inline-snapshot-0.15.0, pretty-1.2.0, requests-mock-1.12.1
collected 1 item                                                                                                                                                              

../../Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_769.py F                                                                                    [100%]
Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_769.py:3 test_foo - AssertionError: assert False  +  where False = isinstance(<ast.Name objec…    [100%]
================================================================================== FAILURES ===================================================================================
__________________________________________________________________________________ test_foo ___________________________________________________________________________________

    def test_foo():
        x = [1]
>       assert x == snapshot(x)

../../Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_769.py:6: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.12/site-packages/inline_snapshot/_inline_snapshot.py:216: in __eq__
    return self == other
.venv/lib/python3.12/site-packages/inline_snapshot/_inline_snapshot.py:267: in __eq__
    self._changes.append(next(it))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <inline_snapshot._adapter.sequence_adapter.ListAdapter object at 0x10391de80>, old_value = [1], old_node = <ast.Name object at 0x103e0abd0>, new_value = [1]

    def assign(self, old_value, old_node, new_value):
        if old_node is not None:
>           assert isinstance(
                old_node, ast.List if isinstance(old_value, list) else ast.Tuple
            )
E           AssertionError: assert False
E            +  where False = isinstance(<ast.Name object at 0x103e0abd0>, <class 'ast.List'>)

.venv/lib/python3.12/site-packages/inline_snapshot/_adapter/sequence_adapter.py:48: AssertionError
```

**Comment by 15r10nk** [comment]
Thank you for the report.

I actually tested logfire with the new inline-snapshot version before the release, but only my branch, where I added the `Is(...)` expressions.

#149 should solve this issue.

I don't know if you are interested in commit where I added the `Is(...)` expressions and used `--inline-snapshot=update` to change some strings and remove some default values. 

https://github.com/pydantic/logfire/compare/main...15r10nk:logfire:inline-snapshot-update-fixes

The resulting code reports no changes for `update` any more, which is also something you asked for.

**Comment by 15r10nk** [comment]
fixed in 0.15.1
